src/main/webapp/script.js

```diff
@@ -108,3 +125,169 @@ function addHiddenPoiFormTrigger() {
+// allows one-day trips.
+function setEndDateValue() {
+  const startDateInput = document.getElementById('inputStartDate');
+  const endDateInput = document.getElementById('inputEndDate');
+  endDateInput.value = startDateInput.value;
+  checkValidInput('inputEndDate');
+}
+
+// Returns true if all location and date inputs are valid; otherwise, false;
+function isLocationDateInputValid() {
+  // Get three input fields.
+  const destinationInput = document.getElementById('inputDestination');
+  const startDateInput = document.getElementById('inputStartDate');
+  const endDateInput = document.getElementById('inputEndDate');
+
+  return destinationInput.classList.contains('is-valid') && 
+    startDateInput.classList.contains('is-valid') &&
+    endDateInput.classList.contains('is-valid');
+}
+
+// Return true if user has submitted a POI; otherwise, false.
+function isPoiSubmitted() {
+  const poiInputs = document.getElementsByClassName('poi-input');
+  return poiInputs.length !== 0;
+}
+
+// If location and date forms have valid inputs, enable next button; otherwise,
+// disable button.
+function checkNextButton() {
+  const toggleStartTripStageButton = document.getElementById('toggle-stage-button');
+
+  // Enable next button if all forms have valid input.
+  if (isLocationDateInputValid()) {    
+    toggleStartTripStageButton.disabled = false;
+  } else {
+    toggleStartTripStageButton.disabled = true;
+  }
+}
+
+// If location and date forms have valid inputs, and there is at least one POI,
+// enable submit button; otherwise, disable button.
+function checkSubmitButton() {
+  const startTripSubmitButton = document.getElementById('submit-calculate-trip');
+
+  // Enable submit button if all forms have valid input.
+  if (isLocationDateInputValid() && isPoiSubmitted()) {    
+    startTripSubmitButton.disabled = false;
+  } else {
+    startTripSubmitButton.disabled = true;
+  }
+}
+
+// If POI text input is valid, enable "Add POIs" button; otherwise, disable button.
+function checkAddPoiButton() {
+  const addPoiButton = document.getElementById('addPoiButton');
+  const inputPoi = document.getElementById('inputPoi');
+
+  // Enable submit button if text input POI is valid.
+  if (inputPoi.value !== '') {
+    addPoiButton.disabled = false;
+  } else {
+    addPoiButton.disabled = true;
+  }
+}
+
+/**
+ * Toggle the start trip input stage, initiated from click of "Next/Back" button.
+ * If toggle "Next/Back" button equals "Next", set location and date inputs to
+ * readonly and show POI input and submit button.
+ * If toggle "Next/Back" button equals "Back", set location and date inputs to 
+ * editable and hide POI input and submit button.
+ */
+function toggleStartTripInputStage() {
+  const toggleStartTripStageButton = document.getElementById('toggle-stage-button');
+
+  if (toggleStartTripStageButton.value === 'Next') {
+    // Display the "Add POIs" form and "Submit" button.
```

Comment:
No action required: Mostly a note-to-self but we should discuss the model-view-controller pattern. Bootsie may know it better than I do, but it's a strategy for separating the state of the webpage from the visual display of the webpage. Mentioning it here because it would involve changing this check, so that instead of looking at the value of an HTML object you'd keep a "model" object in javascript which would keep track of what stage of input the user is in.

MVC might be overkill for this page, so no action requested in this PR, just something we should discuss and consider.